### ISSUE TITLE
Exception in thread "main" java.lang.UnsatisfiedLinkError: no rocksaw in java.library.path

### ISSUE THREAD

**Issue by stodge** [mlaccetti/rocksaw#6]
I added the maven dependency to my POM:

```
<dependency>
	<groupId>com.github.mlaccetti</groupId>
	<artifactId>rocksaw</artifactId>
	<version>1.1.0</version>
	<scope>test</scope>
</dependency>
```

I have a local copy of rocksaw via Maven:

```
ls ~/.m2/repository/com/github/mlaccetti/rocksaw/1.1.0/
m2e-lastUpdated.properties  _remote.repositories  rocksaw-1.1.0.jar  rocksaw-1.1.0.jar.sha1  rocksaw-1.1.0.pom  rocksaw-1.1.0.pom.sha1  rocksaw-1.1.0-sources.jar  rocksaw-1.1.0-sources.jar.sha1
```

My code is fairly simple, and there are no compile time errors:

```
this.socket = new RawSocket();
this.socket.open(PF_INET, getProtocolByName("udp"));
```

But I get a runtime exception:

```
Exception in thread "main" java.lang.UnsatisfiedLinkError: no rocksaw in java.library.path
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1860)
	at java.lang.Runtime.loadLibrary0(Runtime.java:871)
	at java.lang.System.loadLibrary(System.java:1124)
	at com.savarese.rocksaw.net.RawSocket.<clinit>(RawSocket.java:76)
	network.udp.RawUdpPacketSender.start(RawUdpPacketSender.java:27)
	network.udp.RawUdpPacketSender.main(RawUdpPacketSender.java:35)
```

Is there something more I need to, to configure Eclipse?

Eclipse:
Version: 2020-06 (4.16.0)
Build id: 20200615-1200

`Linux: Linux mike 5.7.15-100.fc31.x86_64 #1 SMP Tue Aug 11 17:18:01 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

Thanks

**Comment by mlaccetti** [comment]
The `<scope>test</scope>` should probably be removed for it to work properly; it will only be exposed to tests otherwise.

**Comment by stodge** [comment]
Thanks. I forgot to say that I was writing a unit test, so the scope was correct. I get the exception with or without it.

**Comment by stodge** [comment]
It's trying to load a C library for rockshaw, via JNI. Is there supposed to be a C library (.so) included from Maven? Or do I have to build it myself and install it locally?

**Comment by stodge** [comment]
For anyone reading this; I manually built `rocksaw.so` and copied it to `/usr/lib'. I ran `sudo ldconfig` and it finds the C libary now.

To run/debug an app using raw sockets in Eclipse, assuming the JDK used by Eclipse is at:

`/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.265.b01-1.fc31.x86_64`

```
sudo setcap cap_net_raw,cap_net_admin=eip /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.265.b01-1.fc31.x86_64
```

Add the following to `/etc/ld.so.conf.d/java.conf`:

```
/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.265.b01-1.fc31.x86_64/jre/lib/amd64/jli
/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.265.b01-1.fc31.x86_64/lib/amd64/jli
```

Rebuild the cache:

`sudo ldconfig`

Restart Eclipse and try running the app.
